### PR TITLE
Update use of http2 listen directive to align with deprecation

### PR DESCRIPTION
**Pull request by shaun-nx**
### Proposed changes

As of NGINX 1.25.1, the `http2` option for the [listen](https://nginx.org/en/docs/http/ngx_http_core_module.html#listen) directive has been deprecated.
 
This change updates our main NGINX, VirtualServer and Ingress templates to enable the new [http2](https://nginx.org/en/docs/http/ngx_http_v2_module.html#http2) directive in NGINX.

Change this:
```
listen 443 ssl http2;
```

To this:
```
listen 443 ssl;
http2 on;
```

Resolves https://github.com/nginxinc/kubernetes-ingress/issues/4237

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]
👍🏻

File: internal/configs/version1/template_test.go [hunk 1]
```diff
@@ -665,6 +665,290 @@ func TestExecuteTemplate_ForMainForNGINXPlusWithCustomDefaultHTTPSListenerPort(t
...
+
+	err := tmpl.Execute(buf, mainCfgHTTP2On)
+	t.Log(buf.String())
+
+	if err != nil {
+		t.Fatalf("Failed to write template %v", err)
+	}
+
+	wantDirectives := []string{
+		"http2 on;",
```
Comment (vepatel) [inline internal/configs/version1/template_test.go:718]: can we assert both lines here:
```
wantDirectives := []string{
		"listen 443 ssl;"
                 "http2 on;"
	}
```

File: internal/configs/version1/template_test.go [hunk 2]
```diff
@@ -665,6 +665,290 @@ func TestExecuteTemplate_ForMainForNGINXPlusWithCustomDefaultHTTPSListenerPort(t
...
+	}
+
+	unwantDirectives := []string{
+		"listen 443 ssl http2",
```
Comment (vepatel) [inline internal/configs/version1/template_test.go:722]:
```suggestion
		"listen 443 ssl http2;",
```

File: internal/configs/version1/template_test.go [hunk 3]
```diff
@@ -665,6 +665,290 @@ func TestExecuteTemplate_ForMainForNGINXPlusWithCustomDefaultHTTPSListenerPort(t
...
+		"listen [::]:443 ssl http2",
```
Comment (vepatel) [inline internal/configs/version1/template_test.go:723]:
```suggestion
		"listen [::]:443 ssl http2;",
```

**Review by vepatel — APPROVED** [review]

**Comment by vepatel** [comment]
Hi @hanyouqing this is included in v3.4.0 which is released today 👍🏼

**Pull request merged**